### ISSUE TITLE
didFinishLaunchingWithOptions gets 'clicked_branch_link' = 1 after sharing a link

### ISSUE THREAD

**Issue by contreras2004** [BranchMetrics/ios-branch-deep-linking#857]
When I share a link within my app and after I minimize and reopen the app, the didFinishLaunchingWithOptions receives params like if a link was clicked ONLY after sharing the link to the reminders app. This also happens in the BranchMonsterFactory Example app and it is cross devices. If I share a link to my reminders app and then open the app by clicking directly on the app icon in ANOTHER device, it triggers the click functionality.

**Comment by E-B-Smith** [comment]
I think Apple might have started doing a link preview for the Reminders app which would trigger this.

Thanks for reporting this - I'll look into it and hopefully have a fix soon.

**Comment by contreras2004** [comment]
This is really strange... When I click on a link that was shared in one device, the next device that opens the app will trigger the "clicked_branch_link" and open the shared link data, even when the device is totally different in terms of IOS Version and model. I don't know if this is a session issue or what.

**Comment by E-B-Smith** [comment]
For this:
> When I click on a link that was shared in one device, the next device that opens the app will trigger
> the "clicked_branch_link" 

This is due to a deferred deep link matching confusion - Our back end sees two devices from the same place open the link and decides it's a match.  Which is related to the original question.

-- 

On which version of iOS / device type are you seeing the link re-open after sharing to reminders?

I reproduced this once on iOS 12, but then I upgraded to the latest version of the beta and I can't reproduce it again.

Can you list exact steps in case I'm missing something? This is what I'm doing with my test app:

1. Share a link to the Reminders app.
2. Background the test app.
3. Re-open the test app.
4. What you see: Branch opens the link like it was clicked on.
    What I see: Branch reports "clicked_branch_link=0".

Also, are you testing in a place where a lot of devices share a single IP address, especially if they are testing the same app?  That could be confusing the matching.

**Comment by E-B-Smith** [comment]
I can reproduce this now.  I have a fix that will be out early next week.

**Comment by E-B-Smith** [comment]
This is fixed in release 0.25.5.
